base/hamt.jl

```diff
@@ -75,6 +75,13 @@ function HAMT{K,V}(k::K, v) where {K, V}
 end
 HAMT(k::K, v::V) where {K, V} = HAMT{K,V}(K, V)
 
+function hash(x)::UInt
+    objectid(x)
+end
+function hash(x, h::UInt)::UInt
+    hash(x) - 3h
+end
```

Comment:
```suggestion
hash(x) = objectid(x)::UInt
hash(x, h::UInt) = hash(x) - 3h
```
would avoid calling `convert` which otherwise may lead to invalidations in worse cases.